src/unix/core.c

```diff
@@ -1102,3 +1102,52 @@ int uv_os_homedir(char* buffer, size_t* size) {
 
   return 0;
 }
+
+
+int uv_os_tmpdir(char* buffer, size_t* size) {
+  char* buf;
+  size_t len;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+#define CHECK_ENV_VAR(name) do {                                               \
+  buf = getenv(name);                                                          \
+  if (buf != NULL)                                                             \
+    goto return_buffer;                                                        \
+} while (0)
+
+  /* Check the TMPDIR, TMP, TEMP, and TEMPDIR environment variables in order */
+  CHECK_ENV_VAR("TMPDIR");
+  CHECK_ENV_VAR("TMP");
+  CHECK_ENV_VAR("TEMP");
+  CHECK_ENV_VAR("TEMPDIR");
+
+#undef CHECK_ENV_VAR
+
+  /* No temp environment variables defined */
+  #if defined(__ANDROID__)
+    buf = "/data/local/tmp";
+  #else
+    buf = "/tmp";
+  #endif
+
+return_buffer:
+  len = strlen(buf);
+
+  if (len >= *size) {
+    *size = len;
+    return -ENOBUFS;
+  }
+
+  /* The returned directory should not have a trailing slash. */
+  if (len > 1 && buf[len - 1] == '/') {
+    buf[len - 1] = '\0';
```

Comment:
This modifies the environment (bad) and isn't really necessary because you're copying out the string a few lines down.  I'd modify the logic so you copy out one byte fewer if the string ends in a slash and always zero-terminate manually.